### PR TITLE
Add configuration UI

### LINKED ISSUE #10: Feature request: support additional adjustments to bolding method
Bionic Reading has some features which I think could be useful here and not too hard to implement:

- "fixation" - what percent of the word is bolded
- "saccade" - how often to apply the bolding (so 0 would be every word, 1 is every other word, etc)
- "opacity" - the opacity of the bolded text, seems like this could be a useful substitute for font weight if not supported (see below)

Although bionic reading doesn't support this, it also would be nice if it could support adjusting the font weight of the bolding. Maybe this could automatically substitute a variable weight font if the font the text is in doesn't support it? (ideally this behavior would be a configurable parameter)

Great job on this extension by the way, I've already used it a bit and it works really well!

### PR DESCRIPTION
**Pull request by Merlin04**
This is a work-in-progress PR to add the features I suggested in #10. It's quite a big change to the codebase so I thought I'd make a draft PR so you can see the progress. It now uses rollup for building, web-ext for testing (automatically rebuilds/reloads the extension in your browser when you make a change), and uhtml/water.css for the UI.

### CODE REVIEW

**Comment by Merlin04** [comment]
I've finished implementing the changes, let me know your thoughts!

**Comment by MichaelZhao21** [comment]
I will take a look at it tonight

File: extension/manifest.json [hunk 1]
```diff
@@ -2,27 +2,40 @@
...
     "browser_action": {
         "default_icon": "icons/32.png",
         "default_title": "Turn BiReader On"
     },
-
     "background": {
-        "scripts": ["browser-polyfill.js", "background.js"]
+        "scripts": [
+            "background.js"
+        ]
```
Comment (MichaelZhao21) [inline extension/manifest.json:17]: These lines don't need to be reformatted

File: extension/manifest.json [hunk 2]
```diff
@@ -2,27 +2,40 @@
...
     "content_scripts": [
         {
-            "matches": ["<all_urls>"],
-            "js": ["reader.js"]
+            "matches": [
+                "<all_urls>"
+            ],
+            "js": [
+                "reader.js"
+            ]
```
Comment (MichaelZhao21) [inline extension/manifest.json:26]: These lines also don't need to be reformatted as the array can fit in one line. Having new lines for a single array element decreases readability

**Review by MichaelZhao21 — CHANGES_REQUESTED** [review]
Overall, the settings work really well! However, is it really necessary to use `water.css` or the `uhtml` library to add extra code to the website? Although it does look nice, it might be a better idea to embed that sort of configuration page in the built in [browser action popup](https://developer.chrome.com/docs/extensions/reference/action/#popup). I can merge what you have for now and look into how to integrate your popup into the native browser action.

Another issue is that you should remove the `extension\` build directory from version control. As for the assets, you would need to copy those into the build directory at build time, but that should be easy with rollup (or you could just use `fs`). Have you ever used [esbuild](https://esbuild.github.io/)? It's much simpler and potentially faster than rollup (though I can mess with that at a later time as well).

**Comment by MichaelZhao21** [comment]
One last thing: I've never actually used `web-ext` before, but I am getting the following error when I use `yarn run dev:ff`. I believe it's because I develop in WSL2, which causes the extension to not be able to find the `firefox` Windows executable. I guess this isn't possible to fix directly due to the nature of `web-ext`, but perhaps an alternative manual build/deployment script could be a useful future feature. Currently, I just have to run the build and manually upload the generated zip file.

```
yarn run v1.22.17
$ NODE_ENV=development concurrently "yarn build:ff --watch" "web-ext run -t firefox-desktop -s extension"
[1] Applying config file: ./package.json
$ yarn manifest firefox && IS_FIREFOX=true yarn compile --watch
[1] Running web extension from /home/mikey/temp/bireader/extension
$ esno scripts/generateManifest.js firefox
[1]
[1] Error: not found: firefox
[1]     at F (/home/mikey/temp/bireader/node_modules/fx-runner/node_modules/which/which.js:63:19)
[1]     at E (/home/mikey/temp/bireader/node_modules/fx-runner/node_modules/which/which.js:72:29)
[1]     at /home/mikey/temp/bireader/node_modules/fx-runner/node_modules/which/which.js:81:16
[1]     at /home/mikey/temp/bireader/node_modules/fx-runner/node_modules/isexe/index.js:44:5
[1]     at /home/mikey/temp/bireader/node_modules/fx-runner/node_modules/isexe/access.js:8:5
[1]     at FSReqCallback.oncomplete (node:fs:188:23)
[1]
[1] web-ext run -t firefox-desktop -s extension exited with code 1
[0] Wrote manifest to extension/manifest.json
$ rollup -c --watch
[0] rollup v2.75.6
[0] bundles src/contentScript/reader.js → extension/reader.js...
[0] (!) If you do not supply "output.name", you may not be able to access the exports of an IIFE bundle.
[0] (!) Circular dependency
[0] src/contentScript/reader.js -> src/lib/ui.js -> src/contentScript/reader.js
[0] created extension/reader.js in 394ms
[0] bundles src/background/background.js → extension/background.js...
[0] created extension/background.js in 88ms
```

File: extension/manifest.json (hunk 1 above)
Comment (Merlin04) [reply extension/manifest.json:17]: This file is actually autogenerated from `scripts/generateManifest.js` (which contains a function to return a manifest object based on the browser, so it doesn't have to do patching stuff) - right now the content of the file is produced with `JSON.stringify(/* manifest object */, null, 4)`, I could make it not format the file and put it all in one line if that would be better?

**Comment by Merlin04** [comment]
Thanks so much for the feedback!

I thought of using the browser action popup, but I wanted to keep the browser action just being a toggle button - that way it still only takes a single click to enable it. If you think it would be better it shouldn't be too hard to refactor it to move it into the browser action popup, you'd just need to add an enable button and make the boldify function in the popup send a message to the content script to run with the given parameters.

The `extension` folder was meant to hold all the static files of the extension, a bit like [vitesse-webext](https://github.com/antfu/vitesse-webext) - then the javascript just compiles into that folder, and web-ext can package it nicely and put it in `web-ext-artifacts`. I could make a `build` folder that the compiled output goes into and assets are copied into, but that adds some complexity - I could also just gitignore the js files and manifest.json, although not sure if that separates it enough. Either way, definitely agree they shouldn't be included in VC.

I haven't ever used esbuild, I've heard of it though! Right now the build time isn't too long since there's relatively few dependencies, but it could definitely be worth looking into to reduce complexity.

Unfortunately [it seems like WSL isn't supported by `web-ext`](https://github.com/mozilla/web-ext/issues/2108), could you try this and see if it works well enough for you?

```sh
yarn add --dev chokidar-cli
NODE_ENV=development yarn concurrently "yarn build:ff --watch" "chokidar \"extension/**/*\" -c \"yarn package\""
```

This should automatically build it to `web-ext-artifacts/bireader-1.1.zip` whenever a file changes.

**Comment by MichaelZhao21** [comment]
I'll merge the changes you've made so far and create a couple of issues to deal with the concerns I have. After those are addressed, I'll publish another version of the extension!

**Review by MichaelZhao21 — APPROVED** [review]

**Comment by MichaelZhao21** [comment]
@Merlin04 I've made the changes that I mentioned and published the new versions for review on the Chrome & Firefox store, so hopefully they get updated soon. Tagging you in case you were interested in seeing what has changed.

**Pull request merged**